### PR TITLE
adding mandb suppression logic.

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
This PR will bring in changes to CSE installation steps to suppress real-time man-db updates when installing dependencies. After CSE installation completes, a background process is spawned that will trigger man-db to update its indices with newly installed dependencies without delaying termination of the CSE installation process.

**Which issue(s) this PR fixes**:

Fixes #
Currently, when dependencies are installed during CSE execution time, automatic updates are triggered on man-db. This massively increases the time it takes to install each dependency during CSE execution, as it can take man-db a lot of time to update its database for each downloaded dependency.

This PR will prevent man-db updates from slowing down CSE execution by suppressing man-db's auto-update feature and having it update its indices AFTER all dependencies have been installed. This update process will now take place within a background process that does not block CSE execution.

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/install-dependencies.sh [hunk 1]
```diff
@@ -556,3 +559,6 @@ if [[ $OS == $UBUNTU_OS_NAME ]]; then
   # multi-user.target usually start at the end of the boot sequence
   sed -i 's/After=network-online.target/After=multi-user.target/g' /lib/systemd/system/motd-news.service
 fi
+
+echo "Recreating man-db auto-update flag file and kicking off man-db update process.."
+/usr/bin/mandb &
```
Comment (alexeldeib) [inline vhdbuilder/packer/install-dependencies.sh:564]: this is in vhd, doesn't need to be backgrounded

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 2]
```diff
@@ -15,6 +15,19 @@ TELEPORTD_PLUGIN_DOWNLOAD_DIR="/opt/teleportd/downloads"
...
+MAN_DB_AUTO_UPDATE_FLAG_FILEPATH="/var/lib/man-db/auto-update"
+
+removeManDbAutoUpdateFlagFile() {
+    if [ -f "$MAN_DB_AUTO_UPDATE_FLAG_FILEPATH" ]; then
+        rm $MAN_DB_AUTO_UPDATE_FLAG_FILEPATH
+    fi
+}
+
+createManDbAutoUpdateFlagFile() {
+    if [! -f "$MAN_DB_AUTO_UPDATE_FLAG_FILEPATH" ]; then
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_install.sh:27]: need a space after [ and before !

**Review by ganeshkumar5699 — COMMENTED** [review]
It seems like you tested it with the timestamps and I wonder: were you able to measure the time it took to execute the script (until where you added the timestamp), and see any noticeable reduction in execution time?

**Comment by cameronmeissner** [comment]
@ganeshkumar5699 I didn't end up testing with timestamps across the whole CSE execution, but I did add some adhoc timestamping specifically for the man-db update process after all dependencies have been installed. At the moment it looks like this process takes about ~2 seconds on vanilla and ~5 seconds on gpu.

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 3]
```diff
@@ -15,6 +15,19 @@ TELEPORTD_PLUGIN_DOWNLOAD_DIR="/opt/teleportd/downloads"
...
+
+removeManDbAutoUpdateFlagFile() {
+    if [ -f "$MAN_DB_AUTO_UPDATE_FLAG_FILEPATH" ]; then
+        rm $MAN_DB_AUTO_UPDATE_FLAG_FILEPATH
+    fi
+}
+
+createManDbAutoUpdateFlagFile() {
+    if [ ! -f "$MAN_DB_AUTO_UPDATE_FLAG_FILEPATH" ]; then
+        touch $MAN_DB_AUTO_UPDATE_FLAG_FILEPATH
```
Comment (phealy) [inline parts/linux/cloud-init/artifacts/cse_install.sh:28]:
```suggestion
        touch $MAN_DB_AUTO_UPDATE_FLAG_FILEPATH
```

touch won't error if the file already exists, so it's up to you, but we don't really need to have that inside an if check at all.

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 4]
```diff
@@ -15,6 +15,19 @@ TELEPORTD_PLUGIN_DOWNLOAD_DIR="/opt/teleportd/downloads"
 TELEPORTD_PLUGIN_BIN_DIR="/usr/local/bin"
 KRUSTLET_VERSION="v0.0.1"
 MANIFEST_FILEPATH="/opt/azure/manifest.json"
+MAN_DB_AUTO_UPDATE_FLAG_FILEPATH="/var/lib/man-db/auto-update"
+
+removeManDbAutoUpdateFlagFile() {
+    if [ -f "$MAN_DB_AUTO_UPDATE_FLAG_FILEPATH" ]; then
+        rm $MAN_DB_AUTO_UPDATE_FLAG_FILEPATH
```
Comment (phealy) [inline parts/linux/cloud-init/artifacts/cse_install.sh:22]: if you make this `rm -f` it won't error if the file doesn't exist, so you can skip the check here too.

File: parts/linux/cloud-init/artifacts/cse_install.sh (hunk 3 above)
Comment (phealy) [inline parts/linux/cloud-init/artifacts/cse_install.sh:28]: touch will just update the mtime if the file already exists, so you don't really need an if check.

Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/cse_install.sh:28]: It seems you've uncovered the fact that my bash/linux is a bit rusty ;).

**Review by ganeshkumar5699 — APPROVED** [review]
Lgtm! Congrats on the first PR!

**Review by anujmaheshwari1 — APPROVED** [review]

**Pull request merged**